src/ProductConstructionService/ProductConstructionService.SubscriptionTriggerer/SubscriptionTriggerer.cs

```diff
@@ -26,8 +26,23 @@ public SubscriptionTriggerer(
         _workItemProducerFactory = workItemProducerFactory;
     }
 
-    public async Task CheckSubscriptionsAsync(UpdateFrequency targetUpdateFrequency)
+    public async Task TriggerSubscriptionsAsync(UpdateFrequency targetUpdateFrequency)
     {
+        var workItemProducer = _workItemProducerFactory.CreateProducer<UpdateSubscriptionWorkItem>();
+        foreach (var updateSubscriptionWorkItem in await GetSubscriptionsToTrigger(targetUpdateFrequency))
+        {
+            // TODO https://github.com/dotnet/arcade-services/issues/3811 add some kind of feature switch to trigger specific subscriptions
+            //await workItemProducer.ProduceWorkItemAsync(updateSubscriptionWorkItem);
```

Comment:
Can we uncomment this, un-ignore the tests and just register a no-op workitem producer in the project so that it just does not send anything? Otherwise, we might break this again without knowing.

Reply:
Added FakeWorkItemProducer that doesn't do anything, enabled tests